src/main/scala/is/hail/stats/BaldingNicholsModel.scala

```diff
@@ -42,7 +42,7 @@ object BaldingNicholsModel {
       if (f <= 0d || f >= 1d)
         fatal(s"F_st values must satisfy 0.0 < F_st < 1.0, got $f"))
 
-    val nPartitions = nPartitionsOpt.getOrElse(Math.max(nSamples * nVariants / 1000000, 8))
+    val nPartitions = nPartitionsOpt.getOrElse(Math.max((nSamples.toLong * nVariants / 1000000).toInt, 8))
```

Comment:
I kind of prefer:

`nSamples * (nVariants / 1000000)`, since `nVariants` should almost always be the enormous value and it doesn't change the type. I suppose some day we will have 10 billion humans sequenced, but that's pretty far away.

Yeah, especially because its the intermediate product that's too big, not either one of the values.